keanu-project/src/main/java/io/improbable/keanu/algorithms/Statistics.java

```diff
@@ -0,0 +1,39 @@
+package io.improbable.keanu.algorithms;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+public class Statistics {
+
+    private Map<Enum, List<Double>> statistics = new HashMap<>();
+
+    public Statistics(Enum[] keys) {
```

Comment:
Should we perhaps take a "length hint" in here to allow us to pre-allocate a sensible sized set of ArrayLists?  Obviously this won't work for streaming samples, but where we know up front how many samples we're generating etc - this might help us make a first guess at how many stats are coming in and stop us having to reallocate and copy stuff around a bit?

Reply:
Honestly I think that for this debugging tool it's better to keep it simple. If the debug statistics becomes something that people begin to use on every single run then yes, I think we should begin doing stuff like this